Nos/Views/Profile/Edit/CreateUsernameWizard/PickYourUsernameSheet.swift

```diff
@@ -114,31 +114,9 @@ struct PickYourUsernameSheet: View {
     }
 }
 
-fileprivate class UsernameObserver: ObservableObject {
-
-    @Published
-    var debouncedText = ""
-
-    @Published
-    var text = ""
-
-    private var subscriptions = Set<AnyCancellable>()
-
-    init() {
-        $text
-            .removeDuplicates()
-            .filter { $0.count >= 3 }
-            .debounce(for: .milliseconds(200), scheduler: RunLoop.main)
-            .sink { [weak self] value in
-                self?.debouncedText = value
-            }
-            .store(in: &subscriptions)
-    }
-}
-
 fileprivate struct UsernameTextField: View {
 
-    @StateObject var usernameObserver: UsernameObserver
+    @State var usernameObserver: TextDebouncer
```

Comment:
```suggestion
    var usernameObserver: TextDebouncer
```

Reply:
Thanks, @mplorentz. What you're saying matches Apple's [migration guide](https://developer.apple.com/documentation/swiftui/migrating-from-the-observable-object-protocol-to-the-observable-macro), but it doesn't seem to work here. In their example case, the object is passed in to the view (not created by the view like we're doing here).

When I remove `@State`, changes to the text field do not trigger the verification. When I add it back, it works. I asked ChatGPT about it, and it agreed with what you said: that it didn't need `@State`. But then again, I also got some results from ChatGPT that failed to compile. ¯\_(ツ)_/¯ 

The `DiscoverContentsView` change you suggested works; I believe because, like you said, the lifetime is managed by the parent (as a `@State var`). However in this case, we want the lifetime of the `TextDebouncer` to be tied to the lifetime of the view, and it is not managed by the parent, but by the view itself. I also tried `@Bindable`, which I think might be correct for the `usernameObserver` property inside `UsernameTextField`, but it still doesn't work if you don't annotate the property in `PickYourUsernameSheet` with `@State`.